### PR TITLE
fix: fix injection invalid bug by add inject task to handler inject when queue was empty

### PR DESCRIPTION
**Pull request by aoiasd**
relate: https://github.com/milvus-io/milvus/issues/31548

### CODE REVIEW

**Comment by aoiasd** [comment]
rerun ut

File: internal/datanode/flush_manager.go [hunk 1]
```diff
@@ -152,111 +151,108 @@ func (q *orderFlushQueue) getFlushTaskRunner(pos *msgpb.MsgPosition) *flushTaskR
...
-			q.tailMut.Unlock()
-		case <-h.done:
-			return
+func (h *injectTask) waitFinish() {
+	<-h.startSignal
+	var postInjection postInjectionFunc
+	select {
+	case injection := <-h.injectCh:
+		// notify injected
+		injection.injectOne()
```
Comment (xiaofan-luan) [inline internal/datanode/flush_manager.go:243]: merge injection.injectOne and ok := <-injection.injectOver into one function

File: internal/datanode/flush_manager.go [hunk 2]
```diff
@@ -96,18 +96,20 @@ var _ flushManager = (*rendezvousFlushManager)(nil)
...
+	// protect postInjection
 	injectMut     sync.Mutex
-	runningTasks  int32
+	injectCh      chan *taskInjection
 	postInjection postInjectionFunc
+
+	// protect task create and close
+	taskMut      sync.Mutex
+	runningTasks int32
+	tailCh       chan struct{}
```
Comment (xiaofan-luan) [inline internal/datanode/flush_manager.go:112]: tailCh need to be threadsafe?
waitForAllFlushQueue seems not holding taskMut lock

**Comment by xiaofan-luan** [comment]
there is still issues on waitForAllFlushQueue code.

1. waitForAllFlushQueue is not thread safe
2. dropping code is very complicated, I'm not saying there must be a bug but the complexity make me feel bad

**Pull request merged**